h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -910,4 +928,112 @@ private void cleanUpModelsCVPreds() {
         model.deleteCrossValidationPreds();
     }
   }
+
+
+  protected AutoBuffer writeAutoML(AutoBuffer ab) {
+    try (PersistenceContext pc = PersistenceContext.begin()) {
+      ab.put(_key);
+      ab.put(_buildSpec);
+      ab.putKey(_leaderboard._key);
+      ab.putKey(_eventLog._key);
+//      ab.putKey(_trainingFrame._key);
+      for (Model m : leaderboard().getModels()) {
+        PersistenceContext.putKey(ab, m._key);
+        if (_buildSpec.build_control.keep_cross_validation_predictions)
+          for (Key k : m._output._cross_validation_predictions)
+            PersistenceContext.putKey(ab,k);
+        if (_buildSpec.build_control.keep_cross_validation_models)
+          for (Key k : m._output._cross_validation_models)
+            PersistenceContext.putKey(ab,k);
+        if (_buildSpec.build_control.keep_cross_validation_fold_assignment)
+          PersistenceContext.putKey(ab,m._output._cross_validation_fold_assignment_frame_id);
+//        if (m instanceof StackedEnsembleModel)
+//          PersistenceContext.putKey(ab, ((StackedEnsembleModel)m)._output._metalearner._parms._train);
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+    return ab;
+  }
+
+  protected static AutoML readAutoML(AutoBuffer ab, Futures fs) {
+    try (PersistenceContext pc = PersistenceContext.begin()) {
+      AutoML aml = new AutoML(ab.get(), null, ab.get(), false);
+      aml._leaderboard = (Leaderboard) ab.getKey(fs);
+      aml._eventLog = (EventLog) ab.getKey(fs);
+//      aml._trainingFrame = (Frame) ab.getKey(fs);
+      fs.blockForPending();
+      for (Key mk : aml.leaderboard().getModelKeys()) {
+        Model m = (Model) PersistenceContext.getKey(ab, fs, mk);
+        if (aml._buildSpec.build_control.keep_cross_validation_predictions)
+          for (Key k : m._output._cross_validation_predictions)
+            PersistenceContext.loadKey(ab, fs, k);
+        if (aml._buildSpec.build_control.keep_cross_validation_models)
+          for (Key k : m._output._cross_validation_models)
+            PersistenceContext.loadKey(ab, fs, k);
+        if (aml._buildSpec.build_control.keep_cross_validation_fold_assignment)
+          PersistenceContext.loadKey(ab,fs ,m._output._cross_validation_fold_assignment_frame_id);
+//        if (m instanceof StackedEnsembleModel)
+//          PersistenceContext.loadKey(ab, fs, ((StackedEnsembleModel)m)._output._metalearner._parms._train);
+      }
+      DKV.put(aml);
+      return aml;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
```

Comment:
The objects have to be deserialized exactly in the same order as they were serialized => we need to keep track of what was already loaded and in that case skip the loading of that object.